### PR TITLE
Cleanups.

### PR DESCRIPTION
**Pull request by anntzer**
- faulthandler is present in all versions of Python we support; no need
  for `try... except ImportError` anymore.
- LooseVersions can directly be compared to strings (it casts the other
  argument to a LooseVersion).
- Move the gi+cairocffi incompatibility check (duplicated between
  backend_gtk3agg and backend_gtk3cairo) to the common _gtk3_compat.
- Replace HAS_CAIRO_CFFI with checking `cairo.__name__`, which is
  consistent with e.g. the check on `gi.__name__`.
- Inline the (single-used, internal) pilread (I think the new version is
  also more readable as we don't need to handle None as special value).
- Replace HAS_NATGRID by pytest.importorskip.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/__init__.py [hunk 1]
```diff
@@ -1417,19 +1417,15 @@ def tk_window_focus():
 
 
 def _init_tests():
-    try:
+    # CPython's faulthandler since v3.6 handles exceptions on Windows
```
Comment (jklymak) [inline lib/matplotlib/__init__.py]: I'm not fully grokking this, but I thought we were supporting 3.5 for the next release...

Comment (anntzer) [reply lib/matplotlib/__init__.py]: faulthandler exists since 3.3: https://docs.python.org/3.3/library/faulthandler.html

Comment (jklymak) [reply lib/matplotlib/__init__.py]: Yes, but why does the comment make reference to "since v3.6"?

Comment (tacaswell) [reply lib/matplotlib/__init__.py]: It exists on all version (so we don't need the import check), but still need the version check to know if we want to use it on windows or not.

**Pull request merged**